Legacy message assembler unknown function `std::find`

Seeing the following error when building the new API on Ubuntu 24.04.

<details><summary>Build error</summary>

```
emusser@ITAM-866:~/src/crl/multisense/LibMultiSense$ cmake -S . -B build -DBUILD_LEGACY_API=OFF
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Selecting default build type: Release
-- Performing Test HAVE_CONSTEXPR
-- Performing Test HAVE_CONSTEXPR - Success
-- Looking for vasprintf
-- Looking for vasprintf - found
-- Looking for getopt
-- Looking for getopt - found
Missing nlohmann_json. Not building the c++ DeviceInfoUtility
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /home/emusser/src/crl/multisense/LibMultiSense/build
emusser@ITAM-866:~/src/crl/multisense/LibMultiSense$ cmake --build build/ -j15
[  3%] Building CXX object source/Wire/CMakeFiles/MultiSenseWire.dir/source/utility/TimeStamp.cc.o
[  7%] Building CXX object source/Wire/CMakeFiles/MultiSenseWire.dir/source/utility/Exception.cc.o
[ 10%] Linking CXX shared library libMultiSenseWire.so
[ 10%] Built target MultiSenseWire
[ 14%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/channel.cc.o
[ 17%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/info.cc.o
[ 25%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/factory.cc.o
[ 25%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/storage.cc.o
[ 28%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/utilities.cc.o
[ 32%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/utilities.cc.o
[ 35%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/calibration.cc.o
[ 39%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/configuration.cc.o
[ 42%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/ip.cc.o
[ 46%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/message.cc.o
[ 53%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/status.cc.o
[ 53%] Building CXX object source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/udp.cc.o
/home/emusser/src/crl/multisense/LibMultiSense/source/LibMultiSense/details/legacy/message.cc: In member function ‘bool multisense::legacy::MessageAssembler::process_packet(const std::vector<unsigned char>&)’:
/home/emusser/src/crl/multisense/LibMultiSense/source/LibMultiSense/details/legacy/message.cc:278:32: error: ‘find’ is not a member of ‘std’; did you mean ‘bind’?
  278 |             if (auto it = std::find(std::begin(ordered_messages), std::end(ordered_messages), full_sequence_id);
      |                                ^~~~
      |                                bind
gmake[2]: *** [source/LibMultiSense/CMakeFiles/MultiSense.dir/build.make:174: source/LibMultiSense/CMakeFiles/MultiSense.dir/details/legacy/message.cc.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:313: source/LibMultiSense/CMakeFiles/MultiSense.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2

```

</details>

--- a/source/LibMultiSense/details/legacy/message.cc
+++ b/source/LibMultiSense/details/legacy/message.cc
@@ -34,6 +34,7 @@
  *   2025-01-08, malvarado@carnegierobotics.com, IRAD, Created file.
  **/
 
+#include <algorithm>
 #include <inttypes.h>
 #include <limits>
 

